### PR TITLE
Add Domain-Driven Design

### PR DESCRIPTION
**Pull request by heynickc**
[https://github.com/heynickc/awesome-ddd](https://github.com/heynickc/awesome-ddd)

A curated list of Domain-Driven Design (DDD), Command Query Responsibility Segregation (CQRS), Event Sourcing, and Event Storming resources.  It contains books, training courses, video collections, mailing lists and forums, blogs, sample projects, libraries and frameworks, podcasts and interviews, and user groups related to the aforementioned software architecture concepts.

# ALL THE BELOW CHECKBOXES SHOULD BE DONE AND CHECKED AND THE ABOVE PLACEHOLDERS SHOULD BE FILLED IN BEFORE SUBMITTING A PULL REQUEST!

- [x] I have read and understood the [contribution guidelines](https://github.com/sindresorhus/awesome/blob/master/contributing.md) and the [instructions for creating a list](https://github.com/sindresorhus/awesome/blob/master/create-list.md).
- [x] This pull request has a descriptive title. *(For example: `Add Node.js`)*
- The list I added
	- [x] **has been around for at least 20 days,**
	- [x] is a non-generated Markdown file in a GitHub repo,
	- [x] is not a duplicate,
	- [x] only has awesome items with descriptions *(Awesome lists are curations of the best, not of everything)*,
	- [x] contains the [awesome badge](https://github.com/sindresorhus/awesome/blob/master/awesome.md#awesome-badge) on the right side of the list heading,
	- [x] has a Table of Contents section named `Contents` as the first section,
	- [x] has an [appropriate license](https://github.com/sindresorhus/awesome/blob/master/awesome.md#choose-an-appropriate-license) *(Meaning something like CC0, NOT a code licence like MIT, BSD, Apache, etc)*,
	- [x] has [contribution guidelines](https://github.com/sindresorhus/awesome/blob/master/awesome.md#include-contribution-guidelines),
	- [x] has been checked for inconsistent formatting, spelling, and grammar *(This includes starting each link description with an uppercase character and ending it with a period `.`, and consistent naming)*,
	- [x] has been added at the bottom of the appropriate category.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -450,7 +450,7 @@ Check out my [blog](https://blog.sindresorhus.com) and follow me on [Twitter](ht
 - [Userscripts](https://github.com/brunocvcunha/awesome-userscripts) - Enhance your browsing experience.
 - [Pokémon](https://github.com/tobiasbueschel/awesome-pokemon) - Pokémon and Pokémon GO resources and tools.
 - [ChatOps](https://github.com/exAspArk/awesome-chatops) - Managing technical and business operations through a chat.
-
```
Comment (sindresorhus) [inline readme.md]: Don't remove the newline.

File: readme.md [hunk 2]
```diff
@@ -450,7 +450,7 @@ Check out my [blog](https://blog.sindresorhus.com) and follow me on [Twitter](ht
...
+- [Domain-Driven Design](https://github.com/heynickc/awesome-ddd) - DDD, CQRS, Event Sourcing, and Event Storming.
```
Comment (sindresorhus) [inline readme.md]: Would be better if the description was a succinct description of DDD.

**Comment by sindresorhus** [comment]
Feedback:
- There should be a succinct summary at the top about DDD with `Domain-driven design` linkified to https://en.m.wikipedia.org/wiki/Domain-driven_design Also CQRS and Event Sourcing, and Event Storming. All of those should be linkified too. Not everyone coming to your list might have extensive knowledge of DDD.
- Too many quotes `""` in some of the descriptions.
- Items in the `Blogs` section should have a description what they're about and why I should care.
- `InfoQ Blog (tagged "Domain-Driven Design")`. `(tagged "Domain-Driven Design")` should not be part of the link, but rather added as a description. Same with the other similar link there.
- Don't quote descriptions.
- The first character in descriptions should be capitalized.
- Items in the Podcast section should have a ` - ` between the link and date.
- `User Group` items can just be the place name. DDD is implied and moot there.
- I would like to see some links to community resources (Example: https://github.com/sindresorhus/awesome-electron#community). For example: Chat, forums, communities, IRC, etc. http://programmers.stackexchange.com/questions/tagged/domain-driven-design http://stackoverflow.com/questions/tagged/domain-driven-design
- `MassTransit - David Prothero, Chris Patterson. "Distributed Application Framework for .NET."` Not clear what the names mean here. Should be explicit. Description should be first. Applies to all similar items.



> has been checked for inconsistent formatting, spelling, and grammar (This includes starting each link description with an uppercase character and ending it with a period ., and consistent naming),

This is not fully adhered to. Many description do not end in a `.`.

-

The merge conflict needs to be resolved.

**Comment by heynickc** [comment]
Wow, thanks so much for the feedback!  I'll get to work on those things :)

**Comment by sindresorhus** [comment]
Looks good.

**Pull request merged**